Userland/Services/WindowServer/ClientConnection.cpp

```diff
@@ -1115,4 +1115,19 @@ void ClientConnection::set_flash_flush(bool enabled)
     Compositor::the().set_flash_flush(enabled);
 }
 
+void ClientConnection::set_window_parent_from_id(i32 client_id, i32 parent_id, i32 child_id)
```

Comment:
This allows any application to "steal" windows from another application, as long as they can guess a client ID.
It feels like the window owner should let WindowServer know that this is okay before we allow it.

Reply:
Implemented and started using this idea in the last two commits.